proto/extensions.go

```diff
@@ -541,3 +541,47 @@ func RegisterExtension(desc *ExtensionDesc) {
 func RegisteredExtensions(pb Message) map[int32]*ExtensionDesc {
 	return extensionMaps[reflect.TypeOf(pb).Elem()]
 }
+
+func extensionAsLegacyType(v interface{}) interface{} {
```

Comment:
This function and extensionAsStorageType need doc comments explaining what they do.

Reply:
Done. I commented on `Extension.value` had referenced it from `extensionAsLegacyType` and `extensionAsStorageType`.